File: backend/src/main/java/com/campus/assistant/service/rag/Chunker.java
package com.campus.assistant.service.rag;

import java.util.List;
import java.util.Map;

/** 文本分块：按语义 / 固定长度切分，为向量化与检索做准备。 */
public interface Chunker {

    record Chunk(String text, String source, Map<String, Object> metadata) {}

    List<Chunk> split(String text, int chunkSize, int overlap);
}

File: backend/app/rag/chunker.py
class TextChunker:
    """简单按字符切块的占位实现，正式版本可增加重叠窗口、标题层级等策略。"""

    def split(self, text: str, chunk_size: int = 800) -> list[str]:
        return [text[i : i + chunk_size] for i in range(0, len(text), chunk_size)]
